src/NumericalAlgorithms/DiscontinuousGalerkin/NumericalFluxes/NumericalFluxHelpers.hpp

```diff
@@ -0,0 +1,122 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/Variables.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/Protocols.hpp"
+#include "NumericalAlgorithms/DiscontinuousGalerkin/SimpleBoundaryData.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace dg {
+namespace NumericalFluxes {
+
+namespace detail {
+template <typename NumericalFluxType, typename... AllFieldTags,
+          typename... AllExtraTags, typename... Args,
+          typename... PackageFieldTags, typename... PackageExtraTags>
+void package_data_impl(
+    const gsl::not_null<dg::SimpleBoundaryData<tmpl::list<AllFieldTags...>,
+                                               tmpl::list<AllExtraTags...>>*>
+        packaged_data,
+    const NumericalFluxType& numerical_flux_computer,
+    tmpl::list<PackageFieldTags...> /*meta*/,
+    tmpl::list<PackageExtraTags...> /*meta*/, const Args&... args) noexcept {
+  numerical_flux_computer.package_data(
+      make_not_null(&get<PackageFieldTags>(packaged_data->field_data))...,
+      make_not_null(&get<PackageExtraTags>(packaged_data->extra_data))...,
+      args...);
+}
+
+template <typename NumericalFluxType, typename... AllFieldTags,
+          typename... AllExtraTags, typename... NormalDotNumericalFluxTypes,
+          typename... PackageFieldTags, typename... PackageExtraTags>
+void normal_dot_numerical_fluxes_impl(
+    const NumericalFluxType& numerical_flux_computer,
+    const dg::SimpleBoundaryData<tmpl::list<AllFieldTags...>,
+                                 tmpl::list<AllExtraTags...>>&
+        packaged_data_int,
+    const dg::SimpleBoundaryData<tmpl::list<AllFieldTags...>,
+                                 tmpl::list<AllExtraTags...>>&
+        packaged_data_ext,
+    tmpl::list<PackageFieldTags...> /*meta*/,
+    tmpl::list<PackageExtraTags...> /*meta*/,
+    const gsl::not_null<
+        NormalDotNumericalFluxTypes*>... n_dot_num_fluxes) noexcept {
```

Comment:
Output arguments go first.

Reply:
Having these arguments last allows type inference of the `NormalDotNumericalFluxTypes`. Since this is a `detail` helper-implementation it's probably better to keep it like this instead of adding more template helper structs, what do you think?